api/v2/routes_rtfsp_management.go

```diff
@@ -354,3 +301,174 @@ func (api *API) getAuthorizedPrivateNetworks(c *gin.Context) {
 	api.l.Infow("authorized private ipfs network listing requested", "user", username)
 	Respond(c, http.StatusOK, gin.H{"response": networks})
 }
+
+// addUsersToNetwork is used to add a user to the list of authorized users
+// for a given private network.
+func (api *API) addUsersToNetwork(c *gin.Context) {
+	if !dev {
+		Fail(c, errors.New("private networks not supported in production, please use https://dev.api.temporal.cloud"))
+		return
+	}
+	username, err := GetAuthenticatedUserFromContext(c)
+	if err != nil {
+		api.LogError(c, err, eh.NoAPITokenError)(http.StatusBadRequest)
+		return
+	}
+	networkName, exists := c.GetPostForm("network_name")
+	if !exists {
+		FailWithMissingField(c, "network_name")
+		return
+	}
+	users, exists := c.GetPostFormArray("users")
+	if !exists {
+		FailWithMissingField(c, "users")
+		return
+	}
+	network, err := api.nm.GetNetworkByName(networkName)
+	if err != nil {
+		api.LogError(c, err, eh.NetworkSearchError)(http.StatusInternalServerError)
+		return
+	}
+	if err := api.isNetworkOwner(networkName, username); err != nil {
+		api.LogError(c, err, eh.PrivateNetworkAccessError)(http.StatusUnauthorized)
+		return
+	}
+	// make sure the user accounts exist
+	for _, user := range users {
+		if _, err := api.um.FindByUserName(user); err != nil {
+			api.LogError(c, err, eh.UserSearchError)(http.StatusInternalServerError)
+			return
+		}
+	}
+	// combine both the currrent list of authorized users, and the list of users to add
+	network.Users = append(network.Users, users...)
+	// update the users field of the database model only
+	if err := api.nm.UpdateNetworkByName(networkName, map[string]interface{}{"users": network.Users}); err != nil {
+		api.LogError(c, err, "failed to update authorized users for network")
+		return
+	}
+	for _, user := range users {
+		if err := api.um.AddIPFSNetworkForUser(user, networkName); err != nil {
+			api.LogError(c, err, "failed to update network for user")(http.StatusInternalServerError)
+			return
+		}
+	}
+	Respond(c, http.StatusOK, gin.H{"response": "authorized user list updated"})
+}
+
+// removeUsersFromNetwork is used to remove a user from being able
+// to access a network.
+func (api *API) removeUsersFromNetwork(c *gin.Context) {
```

Comment:
`api` exposes a huge number of functions 😱 

something I'm hoping to change with V3! but for now all good 🚀

Reply:
Agreed haha, that's one of the things I'm looking forward to about the V3 API you're working on :D